protocol-units/settlement/mcr/contracts/src/staking/MovementStaking.sol

```diff
@@ -63,23 +63,26 @@ contract MovementStaking is
 
         // roll over from 0 (genesis) to current epoch by block time
         currentEpochByDomain[domain] = getEpochByBlockTime(domain);
+        EnumerableSet.AddressSet storage attesters = attestersByDomain[domain];
 
-        for (uint256 i = 0; i < attestersByDomain[domain].length(); i++) {
-            address attester = attestersByDomain[domain].at(i);
+        for (uint256 i = 0; i < attesters; i++) {
+            address attester = attesters.at(i);
 
-            for (uint256 j = 0; j < custodiansByDomain[domain].length(); j++) {
-                address custodian = custodiansByDomain[domain].at(j);
+            // for every custodian
+            EnumerableSet.AddressSet storage custodians = custodiansByDomain[
+                domain
+            ];
+            for (uint256 j = 0; j < custodians; j++) {
+                address custodian = custodians.at(j);
 
                 // for every delegatee of the attester
-                for (
-                    uint256 k = 0;
-                    k < delegatorsByAttesterByDomain[domain][attester].length();
-                    k++
-                ) {
-                    // todo: can this be replaced with _rollOverAttester?
-                    address delegatee = delegatorsByAttesterByDomain[domain][
+                EnumberableSet.AddressSet
+                    storage delegatees = delegatorsByAttesterByDomain[domain][
```

Comment:
Looks much better, aesthentically and I assume also gas-wise. I'm not an expert on Solidity, though.